### ISSUE TITLE
doc, example, anything?

### ISSUE THREAD

**Issue by connaryscott** [coiouhkc/rundeck-hybrid-login#1]
an example might be nice on how this might actually work

<pre>
ERROR BootStrap: using rdeck.base system property: /var/lib/rundeck
INFO  BootStrap: loaded configuration: /etc/rundeck/framework.properties
INFO  BootStrap: RSS feeds enabled
2014-06-13 11:56:24.069:/rundeck:INFO:  Initializing Spring FrameworkServlet 'grails'
2014-06-13 11:56:24.085::INFO:  Started SocketConnector@0.0.0.0:4440
Unable to use direct char[] access of java.lang.String
java.lang.NoSuchFieldException: count
        at java.lang.Class.getDeclaredField(Class.java:1948)
        at org.codehaus.groovy.grails.web.util.StringCharArrayAccessor.<clinit>(StringCharArrayAccessor.java:53)
        at org.codehaus.groovy.grails.web.util.StreamCharBuffer.readAsString(StreamCharBuffer.java:531)
        at org.codehaus.groovy.grails.web.util.StreamCharBuffer.toString(StreamCharBuffer.java:552)
        at java_lang_CharSequence$toString.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:40)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:120)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callSafe(AbstractCallSite.java:83)
        at org.codehaus.groovy.grails.plugins.web.taglib.SitemeshTagLib$_closure6.doCall(SitemeshTagLib.groovy:179)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
</pre>

**Comment by coiouhkc** [comment]
Hi Chuck, which version of Rundeck are you using? 
Initially the extension have been written for 1.4.2 a while ago. Another question - is Rundeck working fine without the rundeck-hybrid-login plugin?
Generally, rundeck-hybrid-login uses JettyCachingLdapLoginModule for authentication and PropertyFileLoginModule for authorization (dummy passwords). 
the configuration file is then the one of the JettyCachingLdapLoginModule with single additional line for specifying the path to the RolePropertyFile:
file=${path_to_file}
A former colleague of mine forked and updated the extension for newer Rundeck version - please check https://github.com/rkrombho/rundeck-hybrid-login.

**Comment by connaryscott** [comment]
Hi,

I am  using rundeck v1.6, specifically:

<pre>
[chuck@wsgc-dev-centos65-1 rundeck]$ rpm -q rundeck
rundeck-1.6.2-1.5.GA.noarch
</pre>

I found that the "user.dir" property must be defined so that the realm.properties file would even be loaded, in my case, user.dir must be set to /etc/rundeck  

<pre>
[chuck@wsgc-dev-centos65-1 rundeck]$ rpm -q rundeck-config -l |grep realm.properties
/etc/rundeck/realm.properties
</pre>

My slightly modified  config file is here, where I changed:

com.dtolabs.rundeck.jetty.jaas.JettyCachingLdapLoginModule 
to
org.abratuhi.rundeck.jetty.jaas.JettyCachingHybridLoginModule

<pre>
[chuck@wsgc-dev-centos65-1 rundeck]$ cat /etc/rundeck/jaas-activedirectory.conf
activedirectory {
   org.abratuhi.rundeck.jetty.jaas.JettyCachingHybridLoginModule sufficient
   debug="true"
   contextFactory="com.sun.jndi.ldap.LdapCtxFactory"
   providerUrl="ldap://rkldap.notshown.com:389"
   bindDn="notshown"
   bindPassword="notshown"
   authenticationMethod="simple"
   forceBindingLogin="true"
   userBaseDn="DC=notshown,DC=com"
   userRdnAttribute="cn"
   userIdAttribute="sAMAccountName"
   userPasswordAttribute="unicodePwd"
   userObjectClass="user"
   roleBaseDn="DC=notshown,DC=com"
   roleNameAttribute="cn"
   roleMemberAttribute="member"
   roleObjectClass="group"
   cacheDurationMillis="300000"
   reportStatistics="true";
   org.mortbay.jetty.plus.jaas.spi.PropertyFileLoginModule required
   debug="true"
   file="/etc/rundeck/realm.properties";
};
</pre>

I will take note of your RolePropertyFile and perhaps remove the additional PropertyFileLoginModule configuration and perhaps then the user.dir property will not be needed.   Not sure if this will help, will advise shortly.

We are still evaluating rundeck v2, so 1.6 is my current situation.

Thanks for the forked location, i can check that as well.
